components/main/Desktop/ExportButton.jsx

```diff
@@ -48,22 +49,60 @@ function ExportButton({ filters }) {
       .map(v => `'${v.typeName}'`)
       .join(', ');
 
-    // in the case user chooses one neighborhood or all are selected + dates and status
-    const neighborhoodDataQuery = `SELECT * FROM requests_2024 
-      WHERE CreatedDate >= '${filters.startDate}' 
-      AND CreatedDate < '${filters.endDate}'
-      ${requestStatusFilter !== '' ? ` AND Status='${requestStatusFilter}'` : ''}
-      ${filters.councilId !== null ? ` AND NC='${filters.councilId}'` : ''} 
-      AND RequestType IN (${formattedRequestTypes});`;
-
-    const groupedAddressQuery = `SELECT Address, COUNT(*) AS NumberOfRequests FROM requests_2024
-      WHERE CreatedDate >= '${filters.startDate}' 
-      AND CreatedDate < '${filters.endDate}' 
-      AND Status = 'Open' 
-      AND NC = '${filters.councilId}' 
-      AND RequestType IN (${formattedRequestTypes})
-      GROUP BY Address`;
+    const startYear = moment(filters.startDate).year();
+    const endYear = moment(filters.endDate).year();
+
+    const generateQuery = (grouped = false) => {
```

Comment:
I like that you've abstracted the logic for generating these query strings into its own function. I'm wondering if we can improve readability by additionally making a few helper functions like so...

```js
const groupRequestsByAddress(year, startDate, endDate, councilId, types) => 
  `SELECT Address, COUNT(*) AS NumberOfRequests FROM requests_${year}
  WHERE CreatedDate >= '${startDate}' 
  AND CreatedDate < '${endDate}' 
  AND Status = 'Open' 
  AND NC = '${councilId}' 
  AND RequestType IN (${types})
  GROUP BY Address`;
```

```js
const getAllRequests(year, startDate, endDate, types, status='', councilId='') =>
 `SELECT * FROM requests_${year} 
  WHERE CreatedDate >= '${startDate}' 
  AND CreatedDate < '${endDate}'
  ${requestStatusFilter !== '' ? ` AND Status='${status}'` : ''}
  ${filters.councilId !== null ? ` AND NC='${councilId}'` : ''} 
  AND RequestType IN (${types});`;
```

This should simplify the logic of `generateQuery()`. I imagine that the queries that require `UNION_ALL` can look like this:

```js
if (grouped) {
  return `(
    ${groupRequestsByAddress(startYear, filters.startDate, endofStartYear, filters.councilId, types)}
  ) 
  UNION ALL (
    ${groupRequestsByAddress(endYear, startOfEndYear, filters.endDate, filters.councilId, types}
  )
}
// do the same for getAllRequests()
```